Remove ability to delete other users

mbramson opened the pull request:
Prevents users from deleting other users. Forces the user to authorize using canary for the delete action of the user controller, and also hides the delete button for all users in the user index that are not the current user.

Closes #50

mbramson committed:
Remove ability to delete other users

ericworkman commented on web/controllers/user_controller.ex:
```diff
@@ -61,16 +64,16 @@ defmodule Pairmotron.UserController do
-    # Here we use delete! (with a bang) because we expect
-    # it to always work (and if it does not, it will raise).
-    Repo.delete!(user)
+  def delete(conn, %{"id" => _id}) do
```
Is the struct match here needed?

mbramson replied:
Nope! great call!

mbramson committed:
Additionally restrict users from editing others

Restrict users from editing or updating other users.

Remove emails from the show route.

mbramson commented:
So I expanded the scope of this a bit. I additionally restricted users from editing or updating other users. I also hid our emails from the show route, to prevent users from broadcasting their emails to the world.

ericworkman commented on web/templates/user/index.html.eex:
```diff
@@ -19,7 +19,9 @@
       <td class="text-right">
         <%= link "Show", to: user_path(@conn, :show, user), class: "btn btn-default btn-xs" %>
         <%= link "Edit", to: user_path(@conn, :edit, user), class: "btn btn-default btn-xs" %>
```
Should you move the edit button into the if just below?

mbramson replied:
Absolutely. Done.

ericworkman commented:
What happens when a user deletes themselves? They get logged out?

mbramson commented:
That's exactly what happens.

ericworkman commented:
LGTM
